code/modules/mob/holder.dm

```diff
@@ -318,10 +306,40 @@ var/list/holder_mob_icon_cache = list()
 
 				// Add to the cache.
 				holder_mob_icon_cache[cache_key] = mob_icon
-			item_icons[cache_entry] = holder_mob_icon_cache[cache_key]
 
-	// Handle the rest of sync().
-	..(M)
+			var/newstate
+			switch (cache_entry)
+				if (slot_l_hand_str)
+					newstate = "[species_name]_lh"
+				if (slot_r_hand_str)
+					newstate = "[species_name]_rh"
+				if (slot_back_str)
+					newstate = "[species_name]_ba"
+
+			I.Insert(holder_mob_icon_cache[cache_key], newstate)
```

Comment:
What i'm doing here is literally just assembling an icon file which is compatible with contained sprites system.

With a resomi it ends up with the four iconstates for lefthand, righthand, back, and the main image of it body